screens/ProductionScreens/BudgetAddExpense.jsx

```diff
@@ -0,0 +1,181 @@
+import { Text, View, ScrollView } from "react-native";
+import { useContext, useState } from "react";
+import SmallFormButton from "../../components/Form/SmallFormButton.jsx";
+import { AlertContext } from "../../components/Alert/AlertProvider";
+import { icon } from "@fortawesome/fontawesome-svg-core/import.macro";
+import FormField from "../../components/Form/FormField.jsx";
+import ReceiptViewer from "../../components/Budget/ReceiptViewer.jsx";
+import AddRecieptButton from "../../components/Budget/AddRecieptButton.jsx";
+import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome";
+import { faCamera, faImages, faTrash } from "@fortawesome/free-solid-svg-icons";
+import { ref, uploadBytes, getDownloadURL } from "firebase/storage";
+import {
+    launchImageLibraryAsync,
+    launchCameraAsync,
+    requestCameraPermissionsAsync,
+    requestMediaLibraryPermissionsAsync,
+} from "expo-image-picker";
+import { storage } from "../../firebase.config.js";
+
+export default function BudgetAddExpense({ navigation, route }) {
+    const expenseID = Math.floor(Math.random() * 10 ** 20);
+    const [budget, setBudget] = useState("");
+    const [reference, setReference] = useState("");
+    const [description, setDescription] = useState("");
+    const [receiptURI, setReceiptURI] = useState("");
+    const [cost, setCost] = useState("");
+    const { setAlert } = useContext(AlertContext);
+    const storageRef = ref(storage);
+
+    const submitForm = async () => {
+        // TODO: Complete Submit Form Functionality
+        setAlert("Created Budget", "bg-green-500", icon({ name: "circle-check" }));
+        const url = "";
+        if (receiptURI) {
+            try {
+                const blob = receiptURI.blob();
+                const extension = receiptURI.substring(receiptURI.lastIndexOf("."));
+                const snapshot = await uploadBytes(
+                    ref(storageRef, "receipts/" + expenseID + extension),
+                    blob
+                );
+                url = await getDownloadURL(snapshot.ref);
+            } catch (error) {
+                setAlert(
```

Comment:
very much doubt its related to this PR but I've currently got this alert lasting indefinitely

Reply:
Yep so do I ugh

Reply:
Ok. It's something weird with the setTimeout being cancelled when two alerts are added at the same time. Something to sort in the final PR on this page, have removed the problem for now.